optaplanner-core/src/build/revapi-config.json

```diff
@@ -25,68 +25,8 @@
-           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator>",
-           "package": "org.optaplanner.core.config.score.director",
-           "classSimpleName": "ScoreDirectorFactoryConfig",
-           "methodName": "getIncrementalScoreCalculatorClass",
-           "elementKind": "method",
-           "justification": "Score calculators moved package."
-         },
-         {
-           "code": "java.method.parameterTypeParameterChanged",
-           "old": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setEasyScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator>===)",
-           "new": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setEasyScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.api.score.calculator.EasyScoreCalculator>===)",
-           "oldType": "java.lang.Class<? extends org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator>",
-           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.EasyScoreCalculator>",
-           "package": "org.optaplanner.core.config.score.director",
-           "classSimpleName": "ScoreDirectorFactoryConfig",
-           "methodName": "setEasyScoreCalculatorClass",
-           "parameterIndex": "0",
-           "elementKind": "parameter",
-           "justification": "Score calculators moved package."
-         },
-         {
-           "code": "java.method.parameterTypeParameterChanged",
-           "old": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setIncrementalScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator>===)",
-           "new": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setIncrementalScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator>===)",
-           "oldType": "java.lang.Class<? extends org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator>",
-           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator>",
-           "package": "org.optaplanner.core.config.score.director",
-           "classSimpleName": "ScoreDirectorFactoryConfig",
-           "methodName": "setIncrementalScoreCalculatorClass",
-           "parameterIndex": "0",
-           "elementKind": "parameter",
-           "justification": "Score calculators moved package."
-         }
-      ]
+      "_comment": "Changes between 7.44.0.Final and the current branch. These changes are desired and thus ignored.",
+      "ignore": []
```

Comment:
```suggestion
      "ignore": [
       {
           "code": "java.method.returnTypeTypeParametersChanged",
           "old": "method java.lang.Class<? extends org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator> org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::getEasyScoreCalculatorClass()",
           "new": "method java.lang.Class<? extends org.optaplanner.core.api.score.calculator.EasyScoreCalculator> org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::getEasyScoreCalculatorClass()",
           "oldType": "java.lang.Class<? extends org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator>",
           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.EasyScoreCalculator>",
           "package": "org.optaplanner.core.config.score.director",
           "classSimpleName": "ScoreDirectorFactoryConfig",
           "methodName": "getEasyScoreCalculatorClass",
           "elementKind": "method",
           "justification": "Score calculators moved package."
         },
         {
           "code": "java.method.returnTypeTypeParametersChanged",
           "old": "method java.lang.Class<? extends org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator> org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::getIncrementalScoreCalculatorClass()",
           "new": "method java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator> org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::getIncrementalScoreCalculatorClass()",
           "oldType": "java.lang.Class<? extends org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator>",
           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator>",
           "package": "org.optaplanner.core.config.score.director",
           "classSimpleName": "ScoreDirectorFactoryConfig",
           "methodName": "getIncrementalScoreCalculatorClass",
           "elementKind": "method",
           "justification": "Score calculators moved package."
         },
         {
           "code": "java.method.parameterTypeParameterChanged",
           "old": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setEasyScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator>===)",
           "new": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setEasyScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.api.score.calculator.EasyScoreCalculator>===)",
           "oldType": "java.lang.Class<? extends org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator>",
           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.EasyScoreCalculator>",
           "package": "org.optaplanner.core.config.score.director",
           "classSimpleName": "ScoreDirectorFactoryConfig",
           "methodName": "setEasyScoreCalculatorClass",
           "parameterIndex": "0",
           "elementKind": "parameter",
           "justification": "Score calculators moved package."
         },
         {
           "code": "java.method.parameterTypeParameterChanged",
           "old": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setIncrementalScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator>===)",
           "new": "parameter void org.optaplanner.core.config.score.director.ScoreDirectorFactoryConfig::setIncrementalScoreCalculatorClass(===java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator>===)",
           "oldType": "java.lang.Class<? extends org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator>",
           "newType": "java.lang.Class<? extends org.optaplanner.core.api.score.calculator.IncrementalScoreCalculator>",
           "package": "org.optaplanner.core.config.score.director",
           "classSimpleName": "ScoreDirectorFactoryConfig",
           "methodName": "setIncrementalScoreCalculatorClass",
           "parameterIndex": "0",
           "elementKind": "parameter",
           "justification": "Score calculators moved package."
         }
    ]
```

Reply:
This does actually need to be there.

Reply:
Why? It is not needed when I built it locally.

Reply:
The change was merged yesterday. Maybe you didn't have the latest code?

Reply:
I pulled the latest changes today before noon around 11 am.

Reply:
This is incredibly interesting. When I was merging this yesterday, it was necessary. Today, somehow, it is not.
But as you say - if Jenkins is green, everything's OK.

Reply:
@triceo have you tried returning back the <revapi.oldKieVersion>7.39.0.Final</revapi.oldKieVersion> to see if it's still needed as yesterday?

Reply:
Yeah, now we check 7.x against 7.44.0.Final, so there may no longer be the violations.

Reply:
That's the interesting part; the changes @triceo pointed to landed on the 7.x branch only this week:
https://github.com/kiegroup/optaplanner/commit/71b35529bec092f5258356188d1afb62c085f681

Reply:
Found the culprit.
This time I was much quicker than prod guys :) 7.44.0.Final is not released yet and revapi by default skips check if it cannot download the desired artifact. I will configure it to fail (tested locally), so it doesn't happen anymore. Now we just need to wait till next week, so it is released. Thanks for spotting it.